ginhttp/server.go

```diff
@@ -96,9 +96,11 @@ func Middleware(tr opentracing.Tracer, options ...MWOption) gin.HandlerFunc {
 		c.Request = c.Request.WithContext(
 			opentracing.ContextWithSpan(c.Request.Context(), sp))
 
-		c.Next()
+		defer func() {
+			ext.HTTPStatusCode.Set(sp, uint16(c.Writer.Status()))
+			sp.Finish()
```

Comment:
When there is is a panic "c.Writer.Status" will be equal to 200.

This change means we will successfully complete the trace (sp.Finish()), but the response code will be 200 -- indicating the operation was successful. I wonder if that could be a little misleading?

Looking at https://github.com/opentracing-contrib/go-stdlib (which this library is derived from), our current behaviour matches theirs. When there is a panic the span is not finished (sp.Finish() is not called).

How would you feel about proposing a change similar to this to opentracing-contrib/go-stdlib:

```
       defer func() {                                                          
            if recover() != nil {                                                                            
                ext.HTTPStatusCode.Set(                                         
                    sp, uint16(http.StatusInternalServerError))                 
            } else {                                                                                            
                ext.HTTPStatusCode.Set(sp, uint16(sct.status))           
            }                                                                                                       
            sp.Finish()                                                         
        }()  
```

That would mean that they would complete the span with an error if there is a panic. If that (or something similar) was acceptable we could then "backport" the change here.

Reply:
Just noticed https://github.com/opentracing-contrib/go-stdlib !